enterprise_catalog/apps/api/tasks.py

```diff
@@ -737,12 +761,19 @@ def _get_algolia_products_for_batch(
     catalog_uuids_by_key = defaultdict(set)
     customer_uuids_by_key = defaultdict(set)
     catalog_queries_by_key = defaultdict(set)
+    academy_uuids_by_key = defaultdict(set)
+    academy_tags_by_key = defaultdict(set)
 
     catalog_query_uuid_by_catalog_uuid = defaultdict(set)
     customer_uuid_by_catalog_uuid = defaultdict(set)
+    academy_uuids_by_catalog_uuid = defaultdict(set)
+    academy_tags_by_catalog_uuid = defaultdict(set)
 
     # Create a shared convenience queryset to prefetch catalogs for all metadata lookups below.
-    all_catalog_queries = CatalogQuery.objects.prefetch_related('enterprise_catalogs')
+    all_catalog_queries = CatalogQuery.objects.prefetch_related(
+        'enterprise_catalogs',
+        'enterprise_catalogs__academies'
+    )
```

Comment:
```suggestion
    all_catalog_queries = CatalogQuery.objects.prefetch_related(
        'enterprise_catalogs',
        'enterprise_catalogs__academies',
        'enterprise_catalogs__academies__tags',
    )
```

I also wonder how to only fetch the columns we need, like tag titles and academy UUIDs.  I spent 5 minutes trying to look that up but couldn't find a solution.

Reply:
Not too many columns expected in this Tag model. Also, we only expect to have at most 6-7 academies over the design life of this feature.